rts/src/main/java/eta/runtime/RuntimeOptions.java

```diff
@@ -0,0 +1,47 @@
+package eta.runtime;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+public class RuntimeOptions{
+    public static final String RTS_PROPERTIES_PATH = "eta/rts.properties";
+    /*
+     * Loads properties from resource which
+     * may be override by system properties
+     */
+    public static Properties load_properties(){
+        Properties p = new Properties();
+        // Load properties from resource
+        try(InputStream in = RuntimeOptions.class.getClassLoader().getResourceAsStream(RTS_PROPERTIES_PATH)){
+            if(in != null) p.load(in);
+        }catch(IOException e){
+            e.printStackTrace();
+        }
+        Properties sp = System.getProperties();
+        // Set loaded properties to be override by System properties
+        for(String key : sp.stringPropertyNames()){
```

Comment:
You're looping through *all* the System properties with `stringPropertyNames()`. Instead, loop over just the static list of Eta runtime properties.